src/codec.js

```diff
@@ -67,6 +67,48 @@ Int.prototype.valueOf = function() {
 
 codec.Int = Int;
 
+/**
+ * Wherever a row object is returned, it is assigned a "toJSON" function. This
+ * function will create that function in a consistent format.
+ */
+function generateToJSONFromRow(row) {
+  return function(options) {
+    options = options || {
```

Comment:
Merging options with defaults would make adding new options easier. It could be done with `Object.assign` or ES6's spread operator.